tests/validator/validator.cpp

```diff
@@ -1764,3 +1771,47 @@ TEST(Validator, unitComplexCycle)
 
     EXPECT_EQ_ERRORS(expectedErrors, v);
 }
+
+TEST(Validator, duplicatedCellMLUnitsOnCiElement)
+{
+    const std::string math =
+        "<math xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "  <ci cellml:units=\"dimensionless\" cellml:units=\"second\">B</ci>\n"
+        "</math>\n";
+
+    const std::vector<std::string> expectedErrors = {
+        "Attribute cellml:units redefined.",
+        "Could not get a valid XML root node from the math on component 'componentName'.",
```

Comment:
Also don't understand how this error message correlates with the problem of duplicated namespace - not sure how that affects the root node?

Reply:
I think libXml2 takes exception to this really bad xml and gives you nothing back.

Reply:
Could we change the error to be something more general then?  "This was so badly formed I didn't know where to start parsing" instead of pointing the user to an error which doesn't make sense?